specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-06-01-preview/azureADOnlyAuthentication.json

```diff
@@ -0,0 +1,299 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2021-06-01-preview",
+    "title": "SynapseManagementClient",
+    "description": "Azure Synapse Analytics APIs"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "type": "oauth2",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      },
+      "description": "Azure Active Directory OAuth2 Flow"
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Synapse/workspaces/{workspaceName}/azureADOnlyAuthentications/{azureADOnlyAuthentication}": {
```

Comment:
>/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Synapse/workspaces/{workspaceName}/azureADOnlyAuthentications/{azureADOnlyAuthentication} [](http://example.com/codeflow?start=5&length=176)

Need to implement DELETE. If this is a tracked resource, also need to implement tags and PATCH support to update tags (at minimum).

Reply:
AzureAdOnlyAuthentications is not a tracked resource. As far as Delete is concerned, we didn't implement delete as we are not deleting a resource. We are either enabling or disabling AzureADOnlyAuthentication only.

By default, all of the workspaces will have AzureADOnlyAuthentication set as false and user can enable or disable this feature. Having DELETE call will be redundant, as we will not be deleting anything and it will behave in similar fashion as PUT call with value as false

Reply:
If this operation only sets/clears a property value of an existing resource, then modeling it as a resource type isn't what you want. This sounds more like a POST action on the workspaces type.